Thermal Options not working...

If you set Configuration.h like this:

```
//#define THERMAL_PROTECTION_HOTENDS // Enable thermal protection for all extruders
#define THERMAL_PROTECTION_BED     // Enable thermal protection for the heated bed
```

You will get compile time errors in temperature.cpp :

```
sketch\temperature.cpp: In static member function 'static void Temperature::manage_heater()':

temperature.cpp:705: error: 'watch_bed_next_ms' was not declared in this scope

       if (watch_bed_next_ms && ELAPSED(ms, watch_bed_next_ms)) {
```

I think we need to change this block of code to:

```
#if ENABLED(THERMAL_PROTECTION_HOTENDS) && WATCH_TEMP_PERIOD > 0
  int Temperature::watch_target_temp[HOTENDS] = { 0 };
  millis_t Temperature::watch_heater_next_ms[HOTENDS] = { 0 };
#endif

#if ENABLED(THERMAL_PROTECTION_BED) && WATCH_BED_TEMP_PERIOD > 0
  int Temperature::watch_target_bed_temp = 0;
  millis_t Temperature::watch_bed_next_ms = 0;
#endif
```

But even that change is not sufficient to get it to compile.

--- a/Marlin/temperature.cpp
+++ b/Marlin/temperature.cpp
@@ -101,7 +101,7 @@ unsigned char Temperature::soft_pwm_bed;
   millis_t Temperature::watch_heater_next_ms[HOTENDS] = { 0 };
 #endif
 
-#if ENABLED(THERMAL_PROTECTION_HOTENDS) && WATCH_BED_TEMP_PERIOD > 0
+#if ENABLED(THERMAL_PROTECTION_BED) && WATCH_BED_TEMP_PERIOD > 0
   int Temperature::watch_target_bed_temp = 0;
   millis_t Temperature::watch_bed_next_ms = 0;
 #endif

--- a/Marlin/temperature.h
+++ b/Marlin/temperature.h
@@ -105,7 +105,7 @@ class Temperature {
       static millis_t watch_heater_next_ms[HOTENDS];
     #endif
 
-    #if ENABLED(THERMAL_PROTECTION_HOTENDS) && WATCH_BED_TEMP_PERIOD > 0
+    #if ENABLED(THERMAL_PROTECTION_BED) && WATCH_BED_TEMP_PERIOD > 0
       static int watch_target_bed_temp;
       static millis_t watch_bed_next_ms;
     #endif

